### PR TITLE
Check for body in the reponse before to read it

### LINKED ISSUE #35: Error when using `with_mock_dir` and responses with empty body
Thanks for httptest2! I'm using it to test https://github.com/jmaspons/osmapiR and I found an error (v. 1.0.0):

```r
remotes::install_github("jmaspons/osmapiR")  ## Sorry, I have no other calls at hand to reproduce the error
httptest2::with_mock_dir("mock_set_prefs_user", r <- osmapiR::osm_set_preferences_user(key = "test-pref", value = "value"))
#> Error in `resp_body_raw()`:
#> ! Can't retrieve empty body.
#> Run `rlang::last_trace()` to see where the error occurred.
rlang::last_trace()
#> <error/rlang_error>
#> Error in `resp_body_raw()`:
#> ! Can't retrieve empty body.
#> ---
#> Backtrace:
#>      ▆
#>    1. ├─httptest2::with_mock_dir(...)
#>   2. │ ├─httptest2:::with_mock_path(...)
#>   3. │ │ └─base::eval.parent(expr)
#>   4. │ │   └─base::eval(expr, p)
#>   5. │ └─httptest2::capture_requests(expr, simplify = simplify)
#>   6. │   └─base::eval.parent(expr)
#>   7. │     └─base::eval(expr, p)
#>   8. ├─testthat::expect_null(...)
#>   9. │ └─testthat::quasi_label(enquo(object), label, arg = "object")
#>  10. │   └─rlang::eval_bare(expr, quo_get_env(quo))
#>  11. └─osmapiR::osm_set_preferences_user(key = "test-pref", value = "value")
#>  12.   └─httr2::req_perform(req) at osmapiR/R/osmapi_user_data.R:628:3
#>  13.     ├─base::.doTrace(...) at osmapiR/R/osmapi_user_data.R:628:3
#>  14.     │ └─base::eval.parent(exprObj)
#>  15.     │   └─base::eval(expr, p)
#>  16.     │     └─base::eval(expr, p)
#>  17.     └─httptest2::save_response(...)
#>  18.       └─httr2::resp_body_string(response)
#>  19.         └─httr2::resp_body_raw(resp)
#> Run rlang::last_trace(drop = FALSE) to see 2 hidden frames.
```

I think it's missing a check for  `length(response$body)` in https://github.com/nealrichardson/httptest2/blob/cff21ee6e0a38e4b43826493d4afc4862d37a25e/R/capture-requests.R#L128

With `simplify = FALSE` it works because it has the check https://github.com/nealrichardson/httptest2/blob/cff21ee6e0a38e4b43826493d4afc4862d37a25e/R/capture-requests.R#L155

I can do a PR if you want.

### PR DESCRIPTION
**Pull request by jmaspons**
If the response has no body, write and empty file
FIX #35

### CODE REVIEW

**Comment by nealrichardson** [comment]
Thanks. This seems reasonable. Would you mind adding a test that covers this case?

**Comment by jmaspons** [comment]
Yes, I would like to add a test. Could you give me an example about how to simulate and empty response? I've seen that  webfakes is used but I have no experience with it

**Comment by nealrichardson** [comment]
`httr2::response()` makes responses, and as it turns out, with no arguments specified, the body will be empty. So you could test that `save_response()` can write that to a tempfile, and probably should test that we can read it back in with `load_response()`.

**Comment by jmaspons** [comment]
Tests added, @nealrichardson

File: tests/testthat/test-capture-requests.R [hunk 1]
```diff
@@ -42,6 +43,7 @@ test_that("We can record a series of requests (a few ways)", {
     "httpbin.org/image/webp.R-FILE", # The `write_disk` location
     "httpbin.org/put-PUT.json", # Not a GET, but returns 200
     "httpbin.org/response-headers-ac4928.json",
+    "httpbin.org/status/200.txt", # 200 response, so .txt
```
Comment (nealrichardson) [inline tests/testthat/test-capture-requests.R:46]: Let's clarify what this is

```suggestion
    "httpbin.org/status/200.txt", # empty 200 response "text/plain", so .txt
```

**Review by nealrichardson — COMMENTED** [review]
One minor suggestion but otherwise this looks good, thank you! Once I merge #38, can you rebase and add a NEWS bullet for this fix?

**Comment by jmaspons** [comment]
Ready to merge, @nealrichardson

File: NEWS.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # httptest2 1.0.0.9000
 
 * `request` is now removed when saving `httr2_response` objects. In earlier versions of `httr2`, requests were not included in responses, but in httr2 1.0.0, [they were added](https://github.com/r-lib/httr2/pull/359) in order to improve error messages. *If you recorded any responses with httr2 >= 1.0 and httptest2 prior to this version, you may have leaked auth secrets: this would happen if your requests included auth information (as in an `Authentication` header), and the response was saved in a .R file, not simplified to .json or other response-body-only formats. Please inspect your recorded responses and invalidate any tokens that were exposed.*
+* `save_response` now works with `simplify = TRUE` and empty body responses (#37, @jmaspons)
```
Comment (nealrichardson) [inline NEWS.md:4]:
```suggestion
* `save_response()` now works with `simplify = TRUE` when the response body is empty (#37, @jmaspons)
```

**Pull request merged**